karma/database.py

```diff
@@ -0,0 +1,330 @@
+        if not member:
+            member = KarmaMember.add(guild_id, user_id)
+        return member
+
+    @staticmethod
+    def get(guild_id: int, user_id: int) -> Optional[KarmaMember]:
+        query = (
+            session.query(KarmaMember)
+            .filter_by(guild_id=guild_id, user_id=user_id)
+            .one_or_none()
+        )
+        return query
+
+    @staticmethod
+    def get_count(guild_id: int) -> int:
+        count = (
+            session.query(func.count(KarmaMember.user_id))
+            .filter_by(guild_id=guild_id)
+            .scalar()
+        )
+
+        return count
+
+    @staticmethod
+    def get_some(
+        guild_id: int, board: BoardType, order: BoardOrder, limit: int, offset: int
+    ) -> List[KarmaMember]:
+        column = getattr(KarmaMember, board.name)
+
+        if order == BoardOrder.ASC:
+            order_by = column.asc()
+        elif order == BoardOrder.DESC:
+            order_by = column.desc()
+        else:
+            raise ValueError(f"Unsupported BoardOrder {order}.")
+
+        query = (
+            session.query(KarmaMember)
+            .filter_by(guild_id=guild_id)
+            .order_by(order_by)
+            .offset(offset)
+            .limit(limit)
+            .all()
+        )
+
+        return query
+
+    @staticmethod
+    def add(guild_id: int, user_id: int) -> KarmaMember:
+        if KarmaMember.get(guild_id, user_id):
+            raise ValueError(f"Member {user_id} already exists in guild {guild_id}.")
+        member = KarmaMember(guild_id=guild_id, user_id=user_id)
+        session.add(member)
+        session.commit()
+        return member
+
+    @property
+    def value_position(self) -> int:
+        value = (
+            session.query(func.count(KarmaMember.user_id))
+            .filter(KarmaMember.value > self.value)
+            .one()
+        )
+        return value[0] + 1
+
+    @property
+    def given_position(self) -> int:
+        value = (
+            session.query(func.count(KarmaMember.user_id))
+            .filter(KarmaMember.given > self.given)
+            .one()
+        )
+        return value[0] + 1
+
+    @property
+    def taken_position(self) -> int:
+        value = (
+            session.query(func.count(KarmaMember.user_id))
+            .filter(KarmaMember.taken > self.taken)
+            .one()
+        )
+        return value[0] + 1
+
+    def save(self) -> None:
+        session.commit()
+
+    def __repr__(self) -> str:
+        return (
+            f"<KarmaMember idx='{self.idx}' "
+            f"guild_id='{self.guild_id}' user_id='{self.user_id}' "
+            f"value='{self.value}' given='{self.given}' taken='{self.taken}'>"
+        )
+
+    def dump(self) -> Dict[str, int]:
+        return {
+            "guild_id": self.guild_id,
+            "user_id": self.user_id,
+            "value": self.value,
+            "given": self.given,
+            "taken": self.taken,
+        }
+
+
+class KarmaEmoji:
```

Comment:
This class may not be needed, it's only used on some places where we have to use conditions (id/name) anyways.